robotbase/src/main/java/frc/robot/Robot.java

```diff
@@ -44,21 +49,25 @@ public class Robot extends TimedRobot {
   public Robot() {
     swerve = TunerConstants.createDrivetrain();
     coralRunner = new CoralRunner();
+    laterator = new Laterator();
+
     driverControls = new DriverControls();
+    coDriverControls = new CoDriverControls();
 
     m_defaultDrive = new DefaultDrive(
       driverControls::getLeftX,
       driverControls::getLeftY,
       driverControls::getRightX
     );
 
-    this.addPeriodic(
-        () -> {
-          coralRunner.calculateBeamBreaks();
-        },
-        CORAL_RUNNER.SENSOR_PERIODIC_TIME,
-        CORAL_RUNNER.SENSOR_PERIODIC_OFFSET_TIME
-      );
+    addPeriodic(
+      () -> {
+        coralRunner.calculateBeamBreaks();
+        laterator.calculateHallEffect();
+      },
+      SENSOR_PERIODIC.SENSOR_PERIODIC_TIME,
+      SENSOR_PERIODIC.SENSOR_PERIODIC_OFFSET_TIME
+    );
```

Comment:
Been a minute since I've looked at any robot code. Could you explain what this is doing and why we're doing it?